libraries/chain/transaction_context.cpp

```diff
@@ -831,5 +831,23 @@ namespace eosio::chain {
       }
    }
 
+   int64_t transaction_context::set_proposed_producers(vector<producer_authority> producers) {
+      if (producers.empty())
+         return -1; // INSTANT_FINALITY depends on DISALLOW_EMPTY_PRODUCER_SCHEDULE
+
+      EOS_ASSERT(producers.size() <= config::max_producers, wasm_execution_error,
+                 "Producer schedule exceeds the maximum proposer count for this chain");
+
+      trx_blk_context.proposed_schedule_block_num = control.head_block_num() + 1;
+      // proposed_schedule.version is set in assemble_block
+      trx_blk_context.proposed_schedule.producers = std::move(producers);
+
+      return std::numeric_limits<uint32_t>::max();
```

Comment:
Is this meant to be a `uint32_t` instead of `int64_t`? I realize that's how it was before this PR but the types being mismatched here looks sus

Reply:
Yes, that is in accordance to what @arhag requested. https://github.com/AntelopeIO/spring/blob/8ac62810950d7f28835d0a3e7a3bd75f58e5dec4/libraries/chain/include/eosio/chain/webassembly/interface.hpp#L156-L156